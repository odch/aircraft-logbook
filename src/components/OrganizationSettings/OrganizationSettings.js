import React from 'react'
import PropTypes from 'prop-types'
import { isLoaded } from 'react-redux-firebase'
import { Redirect } from 'react-router-dom'
import { injectIntl, intlShape } from 'react-intl'
import withStyles from '@material-ui/core/styles/withStyles'
import Typography from '@material-ui/core/Typography'
import Divider from '@material-ui/core/Divider'
import { organization as organizationShape } from '../../shapes'
import LoadingIcon from '../LoadingIcon'
import DeleteButton from '../DeleteButton'
import OrganizationDeleteDialog from '../OrganizationDeleteDialog/OrganizationDeleteDialog'

const styles = theme => ({
  container: {
    padding: '1em',
    [theme.breakpoints.up(1000 + theme.spacing.unit * 3 * 2)]: {
      width: 1000,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  },
  divider: {
    margin: theme.spacing.unit / 5 + 'em'
  },
  deleteButtonContainer: {
    textAlign: 'center'
  }
})

class OrganizationSettings extends React.Component {
  state = {
    deleteDialogOpen: false
  }

  componentDidMount() {
    this.props.selectOrganization(this.props.organizationId)
  }

  handleDeleteButtonClick = () => {
    this.setState({
      deleteDialogOpen: true
    })
  }

  onDeleteDialogClose = () => {
    this.setState({
      deleteDialogOpen: false
    })
  }

  render() {
    const { organization, classes, deleteOrganization } = this.props

    if (!isLoaded(organization)) {
      return <LoadingIcon />
    }

    if (!organization) {
      return <Redirect to="/" />
    }

    return (
      <div className={classes.container}>
        <Typography variant="title">{organization.id}</Typography>
        <Divider className={classes.divider} />
        <div className={classes.deleteButtonContainer}>
          <DeleteButton
            label={this.props.intl.formatMessage({
              id: 'organizations.delete'
            })}
            onClick={this.handleDeleteButtonClick}
          />
        </div>
        <OrganizationDeleteDialog
          organizationId={organization.id}
          open={this.state.deleteDialogOpen}
          onConfirm={() => deleteOrganization(organization.id)}
          onClose={this.onDeleteDialogClose}
        />
      </div>
    )
  }
}

OrganizationSettings.propTypes = {
  organizationId: PropTypes.string.isRequired,
  organization: organizationShape,
  classes: PropTypes.object.isRequired,
  selectOrganization: PropTypes.func.isRequired,
  deleteOrganization: PropTypes.func.isRequired,
  intl: intlShape
}

export default withStyles(styles)(injectIntl(OrganizationSettings))
