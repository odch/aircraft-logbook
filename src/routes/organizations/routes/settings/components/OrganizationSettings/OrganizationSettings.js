import React from 'react'
import PropTypes from 'prop-types'
import { Redirect } from 'react-router-dom'
import { injectIntl, FormattedMessage } from 'react-intl'
import featureToggles from 'feature-toggles'
import withStyles from '@material-ui/core/styles/withStyles'
import Typography from '@material-ui/core/Typography'
import Divider from '@material-ui/core/Divider'
import Button from '@material-ui/core/Button'
import isLoaded from '../../../../../../util/isLoaded'
import {
  organization as organizationShape,
  intl as intlShape
} from '../../../../../../shapes'
import LoadingIcon from '../../../../../../components/LoadingIcon'
import DeleteButton from '../../../../../../components/DeleteButton'
import CreateMemberDialog from '../../containers/CreateMemberDialogContainer'
import OrganizationDeleteDialog from '../OrganizationDeleteDialog'
import MemberList from '../../containers/MemberListContainer'

const styles = theme => ({
  container: {
    padding: '1em',
    [theme.breakpoints.up(1000 + theme.spacing(3 * 2))]: {
      width: 1000,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  },
  divider: {
    margin: theme.spacing(0.2) + 'em'
  },
  deleteButtonContainer: {
    textAlign: 'center'
  }
})

class OrganizationSettings extends React.Component {
  state = {
    deleteDialogOpen: false
  }

  handleCreateMemberClick = () => {
    this.props.openCreateMemberDialog()
  }

  handleDeleteButtonClick = () => {
    this.setState({
      deleteDialogOpen: true
    })
  }

  onDeleteDialogClose = () => {
    this.setState({
      deleteDialogOpen: false
    })
  }

  render() {
    const {
      organization,
      createMemberDialogOpen,
      classes,
      deleteOrganization
    } = this.props

    if (!isLoaded(organization)) {
      return <LoadingIcon />
    }

    if (!organization) {
      return <Redirect to="/" />
    }

    return (
      <div className={classes.container}>
        <Typography variant="h4" data-cy="organization-title" gutterBottom>
          {organization.id}
        </Typography>
        <Button
          variant="contained"
          color="primary"
          onClick={this.handleCreateMemberClick}
        >
          <FormattedMessage id="organization.settings.createmember" />
        </Button>
        <MemberList organizationId={organization.id} />
        {featureToggles.isFeatureEnabled('organizationsManagement') && (
          <React.Fragment>
            <Divider className={classes.divider} />
            <div className={classes.deleteButtonContainer}>
              <DeleteButton
                label={this.props.intl.formatMessage({
                  id: 'organizations.delete'
                })}
                onClick={this.handleDeleteButtonClick}
                data-cy="organization-delete-button"
              />
            </div>
          </React.Fragment>
        )}
        {createMemberDialogOpen && (
          <CreateMemberDialog organizationId={organization.id} />
        )}
        <OrganizationDeleteDialog
          organizationId={organization.id}
          open={this.state.deleteDialogOpen}
          onConfirm={() => deleteOrganization(organization.id)}
          onClose={this.onDeleteDialogClose}
        />
      </div>
    )
  }
}

OrganizationSettings.propTypes = {
  organization: organizationShape,
  createMemberDialogOpen: PropTypes.bool,
  classes: PropTypes.object.isRequired,
  openCreateMemberDialog: PropTypes.func.isRequired,
  deleteOrganization: PropTypes.func.isRequired,
  intl: intlShape
}

export default withStyles(styles)(injectIntl(OrganizationSettings))
