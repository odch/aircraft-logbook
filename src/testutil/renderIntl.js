import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
