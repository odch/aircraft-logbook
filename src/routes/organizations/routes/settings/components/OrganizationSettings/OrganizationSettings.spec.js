import React from 'react'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import renderIntl from '../../../../../../testutil/renderIntl'
import OrganizationSettings from './OrganizationSettings'

const StartPage = () => <div>Start Page</div>

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('settings', () => {
        describe('components', () => {
          describe('OrganizationSettings', () => {
            it('renders loading icon if organization not loaded', () => {
              const tree = renderIntl(
                <OrganizationSettings
                  organization={undefined}
                  deleteOrganization={() => {}}
                  fetchMembers={() => {}}
                  openCreateMemberDialog={() => {}}
                />
              ).toJSON()
              expect(tree).toMatchSnapshot()
            })

            it('redirects to start page if organization was not found', () => {
              const tree = renderIntl(
                <Router>
                  <Switch>
                    <Route exact path="/" component={StartPage} />
                    <OrganizationSettings
                      organization={null}
                      deleteOrganization={() => {}}
                      fetchMembers={() => {}}
                      openCreateMemberDialog={() => {}}
                    />
                  </Switch>
                </Router>
              ).toJSON()
              expect(tree).toMatchSnapshot()
            })

            it('renders organization settings if loaded', () => {
              const tree = renderIntl(
                <Router>
                  <OrganizationSettings
                    organization={{ id: 'my_org' }}
                    members={[
                      {
                        id: 'member1',
                        firstname: 'Hans',
                        lastname: 'Keller',
                        roles: ['manager']
                      },
                      {
                        id: 'member2',
                        firstname: 'Max',
                        lastname: 'Muster',
                        roles: ['user']
                      }
                    ]}
                    deleteOrganization={() => {}}
                    fetchMembers={() => {}}
                    openCreateMemberDialog={() => {}}
                  />
                </Router>
              ).toJSON()
              expect(tree).toMatchSnapshot()
            })
          })
        })
      })
    })
  })
})
