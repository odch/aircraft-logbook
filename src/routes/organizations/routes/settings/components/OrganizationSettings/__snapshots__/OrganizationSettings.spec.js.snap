// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes settings components OrganizationSettings redirects to start page if organization was not found 1`] = `
<div>
  Start Page
</div>
`;

exports[`routes organizations routes settings components OrganizationSettings renders loading icon if organization not loaded 1`] = `
<div
  className="LoadingIcon-container-4"
>
  <div
    className="MuiCircularProgress-root LoadingIcon-icon-5 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`routes organizations routes settings components OrganizationSettings renders organization settings if loaded 1`] = `
"<div class=\\"injectIntl(OrganizationSettings)-container-1\\">
  <h4 class=\\"MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom\\" data-cy=\\"organization-title\\">my_org</h4><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Mitglied hinzufügen</span><span class=\\"MuiTouchRipple-root\\"></span></button>
  <ul class=\\"MuiList-root MuiList-padding\\">
    <li class=\\"MuiListItem-container\\">
      <div class=\\"MuiListItem-root MuiListItem-secondaryAction\\">
        <div class=\\"MuiListItemText-root MuiListItemText-multiline\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1\\">Keller Hans</span>
          <p class=\\"MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary\\">Manager</p>
        </div>
      </div>
      <div class=\\"MuiListItemSecondaryAction-root\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
    </li>
    <li class=\\"MuiListItem-container\\">
      <div class=\\"MuiListItem-root MuiListItem-secondaryAction\\">
        <div class=\\"MuiListItemText-root MuiListItemText-multiline\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1\\">Muster Max</span>
          <p class=\\"MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary\\">Benutzer</p>
        </div>
      </div>
      <div class=\\"MuiListItemSecondaryAction-root\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
    </li>
  </ul>
  <div class=\\"MuiTablePagination-root\\">
    <div class=\\"MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters\\">
      <div class=\\"MuiTablePagination-spacer\\"></div><span class=\\"MuiTypography-root MuiTablePagination-caption MuiTypography-caption MuiTypography-colorInherit\\">1-2 of 2</span>
      <div class=\\"MuiTablePagination-actions\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z\\"></path></svg></span></button><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z\\"></path></svg></span></button></div>
    </div>
  </div>
  <form>
    <div>
      <div class=\\"MuiFormControl-root MuiTextField-root ExportFlightsForm-datePicker-122 MuiFormControl-marginNormal\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\">Startdatum</label>
        <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd\\" type=\\"text\\" value=\\"01.08.2019\\">
          <div class=\\"MuiInputAdornment-root MuiInputAdornment-positionEnd\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
        </div>
      </div>
      <div class=\\"MuiFormControl-root MuiTextField-root ExportFlightsForm-datePicker-122 MuiFormControl-marginNormal\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\">Enddatum</label>
        <div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd\\"><input aria-invalid=\\"false\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd\\" type=\\"text\\" value=\\"31.08.2019\\">
          <div class=\\"MuiInputAdornment-root MuiInputAdornment-positionEnd\\"><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z\\"></path><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
        </div>
      </div>
    </div>
    <div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary\\" tabindex=\\"0\\" type=\\"submit\\"><span class=\\"MuiButton-label\\">Flüge exportieren</span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
  </form>
  <hr class=\\"MuiDivider-root injectIntl(OrganizationSettings)-divider-2\\">
  <div class=\\"injectIntl(OrganizationSettings)-deleteButtonContainer-3\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\" data-cy=\\"organization-delete-button\\"><span class=\\"MuiButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z\\"></path></svg>Organisation löschen</span><span class=\\"MuiTouchRipple-root\\"></span></button></div>
</div>"
`;
